### PR TITLE
8327255: javac lint warnings: removal, missing-explicit-ctor

### PR DESCRIPTION
**Pull request by kevinrushforth**
This PR updates `build.gradle` to define javac lint options for three different types of java compilation tasks: sdk classes, test classes (including shims), and tool classes (including JSLC). The defaults for these three groups of lint options are specified in `build.gradle`.

We also define three gradle properties that can be used at build time to override the global lint options for each of the three categories of tasks as follows:

* `LINT` - used when compiling sdk tasks : default = "removal,missing-explicit-ctor"
* `TOOL_LINT` - used when compiling tool tasks : default = ""
* `TEST_LINT` - used when compiling test tasks : default = ""

Each property takes a string with a comma-separated list of lint options. For example:

```
gradle -PLINT="deprecation,removal" sdk
```

I provided a build mechanism to add extra per-module lint options in `build.gradle`. None are currently defined, but I tested the logic to show that it can be. I did not provide a way to override the per-module lint options via the command line. Once we start using them, we might consider adding that.

While testing this PR, I ran into a few places where we need to suppress removal warnings for use of our own internal classes that are terminally deprecated, so I filed [JDK-8334143](https://bugs.openjdk.org/browse/JDK-8334143) to track this, and added the missing annotations in this PR.

I also tested with JDK 23 to verify that we are now getting the expected warnings for the use of sun.misc.Unsafe, and filed two bugs --[JDK-8334138](https://bugs.openjdk.org/browse/JDK-8334138) to (temporarily) suppress those warnings in this PR and [JDK-8334137](https://bugs.openjdk.org/browse/JDK-8334137) to eventually fix them by replacing our use of sun.misc.Unsafe.

/issue add JDK-8334143
/issue add JDK-8334138

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issues
 * [JDK-8327255](https://bugs.openjdk.org/browse/JDK-8327255): javac lint warnings: removal, missing-explicit-ctor (**Bug** - P4)
 * [JDK-8334143](https://bugs.openjdk.org/browse/JDK-8334143): Suppress remaining removal warnings for internal methods and classes (**Bug** - P4)
 * [JDK-8334138](https://bugs.openjdk.org/browse/JDK-8334138): Suppress removal warnings for sun.misc.Unsafe memory access methods (**Bug** - P3)

### Reviewers
 * [Ambarish Rapte](https://openjdk.org/census#arapte) (@arapte - **Reviewer**)
 * [Lukasz Kostyra](https://openjdk.org/census#lkostyra) (@lukostyra - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1474/head:pull/1474` \
`$ git checkout pull/1474`

Update a local copy of the PR: \
`$ git checkout pull/1474` \
`$ git pull https://git.openjdk.org/jfx.git pull/1474/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1474`

View PR using the GUI difftool: \
`$ git pr show -t 1474`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1474.diff">https://git.openjdk.org/jfx/pull/1474.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1474#issuecomment-2163814836)

### CODE REVIEW

**Comment by kevinrushforth** [comment]
With the addition of the fixes for JDK-8334143 and JDK-8334138 we are lint clean for removal.

**Comment by kevinrushforth** [comment]
@arapte Can you be the primary reviewer?

File: build.gradle [hunk 1]
```diff
@@ -4071,18 +4115,49 @@ allprojects {
...
+            if (IS_LINT) {
+                def extraLintOpts = project.hasProperty("extraLintOptions") ?
+                                    project.ext.extraLintOptions : ""
+                setupLintOptions(compile, LINT, extraLintOpts);
+
+                // TODO: enable the following once we are lint clean
+                //compile.options.compilerArgs += "-Werror"
+            }
+        } else if (compile.name == "compileJslcJava" ||
+            compile.name == "compileToolJava") {
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply build.gradle:4139]: That was a typo: "compileToolJava" should have been "compileToolsJava". While checking this, I also found that I missed two other tools tasks, "compileDecoraCompilers" and "compilePrismCompilers", so I added them as well.

**Review by arapte — APPROVED** [review]

**Review by lukostyra — APPROVED** [review]
LGTM

**Pull request closed without merging**